cg/meta/clean/demultiplexed_flowcells.py

```diff
@@ -0,0 +1,134 @@
+
+    @property
+    def is_correctly_named(self) -> bool:
+        """checks if the flowcell directory is correctly named: it should end with the flowcell
+        id"""
+        if self._is_correctly_named is None:
+            self._is_correctly_named: bool = self.split_name[-1] == self.identifier
+            if not self._is_correctly_named:
+                LOG.warning("Flowcell name not correctly named!: %s", self.path)
+                return self._is_correctly_named
+        return self._is_correctly_named
+
+    @property
+    def exists_in_statusdb(self) -> bool:
+        """checks if flowcell exists in statusdb"""
+        if self._exists_in_statusdb is None:
+            self._exists_in_statusdb: bool = self.status_db.flowcell(self.id) is not None
+            if not self._exists_in_statusdb:
+                LOG.warning("Flowcell %s does not exist in statusdb!", self.id)
+        return self._exists_in_statusdb
+
+    @property
+    def fastq_files_exist_in_housekeeper(self) -> bool:
+        """Checks if the flowcell has any fastq files in Housekeeper"""
+        if self._fastq_files_exist_in_housekeeper is None:
+            self._fastq_files_exist_in_housekeeper: bool = self.hk_fastq_files.count() > 0
+            if not self._fastq_files_exist_in_housekeeper:
+                LOG.warning("Flowcell %s does not have any fastq files in Housekeeper!", self.id)
+        return self._fastq_files_exist_in_housekeeper
+
+    @property
+    def fastq_files_exist_on_disk(self) -> bool:
+        """Checks if the fastq files that are in Housekeeper are actually present on disk"""
+        if self._fastq_files_exist_on_disk is None:
+            if not self.fastq_files_exist_in_housekeeper:
+                LOG.warning(
+                    "Flowcell %s has no fastq files in Housekeeper, skipping disk check", self.id
+                )
+                self._fastq_files_exist_on_disk = False
+            else:
+                self._fastq_files_exist_on_disk: bool = all(
+                    [Path(fastq_file.path).exists() for fastq_file in self.hk_fastq_files]
+                )
+                if not self._fastq_files_exist_on_disk:
+                    LOG.warning("Flowcell %s has no fastq files on disk!", self.id)
+
+        return self._fastq_files_exist_on_disk
+
+    @property
+    def passed_check(self) -> bool:
+        """indicates if all checks have passed"""
+        if self._passed_check is None:
+            self._passed_check = all(
+                [
+                    self.is_correctly_named,
+                    self.exists_in_statusdb,
+                    self.fastq_files_exist_in_housekeeper,
+                    self.fastq_files_exist_on_disk,
+                ]
+            )
+            if self._passed_check:
+                LOG.info("Flowcell %s has passed all checks, setting flag to True!", self.id)
+        return self._passed_check
+
+    def check_existing_flowcell_directory(self):
```

Comment:
Why do we need both this method and the 'passed_check' above? Seems like the purpose of this one is only to call the former one